ci/generate-workflows.py

```diff
@@ -136,6 +136,18 @@ def __init__(self, name="checkout repo", submodules=True):
         super().__init__(name, action="actions/checkout@v3", params=params)
 
 
+class InstallCrateStep(ActionStep):
+    def __init__(self, crate: str, version=None):
+        params = {"crate": crate}
```

Comment:
It's best to always use `--locked` with `cargo install`.  nextest explicitly doesn't support installing without `--locked`.

If you pass the container name in here, you can use it to set the cache key and that should resolve the glibc issue which arises from falsely sharing cache artifacts.

```suggestion
        params = {"crate": crate, "args": "--locked", "cache-key": container_name}
```

Reply:
Pretty sure the install action defaults to locked, will double check tomorrow, thanks!

Reply:
https://nexte.st/book/installing-from-source.html?highlight=cache#using-a-cached-install-in-ci makes a point of using the old, deprecated `locked: true` flag, which was replaced by `args: --locked`.

Reply:
`baptiste0928/cargo-install@v1` doesn't use `--locked` by default

We are using `v2` here though, which does `--locked` by default, I think merging #3341 might fix caching as it automatically keys by OS, I think it's better to wait for that to be merged before tinkering with this.

Reply:
I will set them explicitely then, and maybe file an issue about containers

Reply:
Used `name` instead of `container_name`

Reply:
Looking at the CI runs I can confirm that it is not necessary to specify `--locked`